### PR TITLE
Using astropy's PyVO to make URL calls.

### LINKED ISSUE #32: Use astroquery.utils.tap rather than making out own URL calls
### Provide a general description of the issue or problem.

https://astroquery.readthedocs.io/en/latest/utils/tap.html

### PR DESCRIPTION
**Pull request by NucleonGodX**
This PR is a part of GSoC-2024, this PR is tackling the issue(#32 ) that is using astroquery's tap plus to do URL calls.

Fixes #32

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -20,6 +20,7 @@ dependencies = [
   "astropy>5.2.0",
   "sunpy[net]>=5.0",
   "requests>=2.28.0",
+  "astroquery"
```
Comment (nabobalis) [inline pyproject.toml:23]:
```suggestion
  "astroquery>=0.4.7"
```

File: tox.ini [hunk 2]
```diff
@@ -34,6 +34,7 @@ set_env =
 deps =
     # For packages which publish nightly wheels this will pull the latest nightly
     devdeps: astropy>=0.0.dev0
+    devdeps: astroquery
```
Comment (nabobalis) [inline tox.ini:37]:
```suggestion
```

File: tox.ini [hunk 3]
```diff
@@ -34,6 +34,7 @@ set_env =
...
     # Packages without nightly wheels will be built from source like this
     devdeps: git+https://github.com/sunpy/sunpy
     devdeps: git+https://github.com/psf/requests
```
Comment (nabobalis) [inline tox.ini:40]:
```suggestion
    devdeps: git+https://github.com/astropy/astroquery.git
```

File: pyproject.toml [hunk 4]
```diff
@@ -20,6 +20,7 @@ dependencies = [
   "astropy>5.2.0",
   "sunpy[net]>=5.0",
   "requests>=2.28.0",
```
Comment (nabobalis) [inline pyproject.toml:22]:
```suggestion
```
Do we need requests still?

Comment (NucleonGodX) [reply pyproject.toml]: No, I'll get rid of it.

File: sunpy_soar/client.py [hunk 5]
```diff
@@ -31,14 +30,14 @@ def search(self, *query, **kwargs):  # NOQA: ARG002
...
         qrt = QueryResponseTable(table, client=self)
-        qrt["Filesize"] = (qrt["Filesize"] * u.byte).to(u.Mbyte).round(3)
-        qrt.hide_keys = ["Data item ID", "Filename"]
+        qrt.hide_keys = ["data_item_id", "Filename", "dimension_index"]
         return qrt
 
     def add_join_to_query(query: list[str], data_table: str, instrument_table: str):
         """
-        Construct the WHERE, FROM, and SELECT parts of the ADQL query.
+        Construct the WHERE, FROM, and SELECT parts of the SQL query.
```
Comment (ebuchlin) [inline sunpy_soar/client.py:40]: I don't understand, before it was URL-encoded ADQL, now it is just ADQL. Transforming the +s to spaces don't make this SQL.

Comment (NucleonGodX) [reply sunpy_soar/client.py:40]: Alrighty, makes sense!

**Comment by ebuchlin** [comment]
Following [this comment](https://github.com/sunpy/sunpy-soar/issues/32#issuecomment-2222646031) on the parent issue #32, [TapPlus](https://astroquery.readthedocs.io/en/latest/utils/tap.html) now recommends to use [PyVO](https://pyvo.readthedocs.io/en/latest/) instead.

Most of line changes in this PR were to make the ADQL queries non-URL-escaped. The transition to PyVO should then not be too difficult as a next step. Shall we do this? (and then change the title of the PR)

**Comment by nabobalis** [comment]
> Most of line changes in this PR were to make the ADQL queries non-URL-escaped. The transition to PyVO should then not be too difficult as a next step. Shall we do this? (and then change the title of the PR)

Yes we should do that.

**Comment by NucleonGodX** [comment]
> Following [this comment](https://github.com/sunpy/sunpy-soar/issues/32#issuecomment-2222646031) on the parent issue #32, [TapPlus](https://astroquery.readthedocs.io/en/latest/utils/tap.html) now recommends to use [PyVO](https://pyvo.readthedocs.io/en/latest/) instead.
> 
> Most of line changes in this PR were to make the ADQL queries non-URL-escaped. The transition to PyVO should then not be too difficult as a next step. Shall we do this? (and then change the title of the PR)

Yep sounds good, also I wanted to ask about one thing, by using these external call methods we restrict the "REQUESTS" to be strictly `doQuery`, as one of the issues raised a few days back suggested to use other REQUESTS methods to query by distance.
So maybe for more diverse queries we would need basic soar payload construction? So till now soar has been using only doQuery method but if we are interested in incorporating other request methods, using anything for external calls maybe won't be a great idea, as I have noticed both TapPlus and PyVo internally sets "REQUESTS" to doQuery only.
I need your thoughts on this. @ebuchlin @nabobalis @hayesla .

**Comment by nabobalis** [comment]
If we need to add support for other queries, we should check upstream with pyvo and then implement it there. That is my immediate thought.

**Comment by NucleonGodX** [comment]
Alright, l'll migrate this PR to using PyVo for now.

**Comment by ebuchlin** [comment]
> So maybe for more diverse queries we would need basic soar payload construction? So till now soar has been using only doQuery method but if we are interested in incorporating other request methods, using anything for external calls maybe won't be a great idea, as I have noticed both TapPlus and PyVo internally sets "REQUESTS" to doQuery only. I need your thoughts on this. @ebuchlin @nabobalis @hayesla .

I don't know, so I've asked in [this comment](https://github.com/sunpy/sunpy-soar/issues/134#issuecomment-2225126549) on the solar distance issue #134.

Going back to requests seems to be a step backwards (although the non-URL encoded ADQL query can still be passed to requests, by using the request payload instead of building the full URL by hand).

**Comment by hayesla** [comment]
OK, so we have discussed this on our weekly calls. And I think the decision was just to punt this for the moment? Am I recalling this correctly?

**Comment by nabobalis** [comment]
> OK, so we have discussed this on our weekly calls. And I think the decision was just to punt this for the moment? Am I recalling this correctly?

Yes that is correct.

**Comment by NucleonGodX** [comment]
> Ok great. I think maybe we should close this PR for the moment, but link it to the original issue. Is that ok with you @NucleonGodX ?

Yep, thats perfectly fine.

**Pull request closed without merging**